holmes.go

```diff
@@ -117,31 +117,23 @@ func (h *Holmes) DisableMemDump() *Holmes {
 	return h
 }
 
-// it won't fit into tiny span since this struct contains point.
-type foo struct {
-	h *Holmes
-}
-
-func finalizerCallback(f *foo) {
+func finalizerCallback(h *Holmes) {
 	// register the finalizer again
-	runtime.SetFinalizer(f, finalizerCallback)
+	runtime.SetFinalizer(h, finalizerCallback)
```

Comment:
oh, no, we should not `h` here since it may be referred in other place, it won't be GC sweep then.

Reply:
oh,I may have misunderstood it.